### DISCUSSION TITLE
Nextjs ISR generated pages lost between deployments

### DISCUSSION THREAD

**Discussion by Hanskrogh** [vercel/next.js#36726] [Ideas]
Hi,

Is there any way to perserve ISR generated pages between deployments, when hosted on vercel?

Vercel/nextjs talks about how ISR strategy can scale to millions of pages. I only find this to be true, if you never make a new deployment, which removes all of your generated pages.

If the ISR page isn't generated, the first request to that pages takes around 5 seconds for us, which is a really poor user experience. 

Any suggestions to what could be done to fix our issue?

**Comment by icyJoseph** [comment]
I don't have a straight answer here, others might, but just a few thoughts :

- Yeah, this is also a problem with optimised images, though I think in Vercel you can make them available to the next (future) build, via https://vercel.com/docs/concepts/deployments/build-step#caching, but I don't think this applies to generated pages. 

- I think this is where a full scale solution comes into play. In one of our large projects we use CloudFront as caching layer, so even between deploys we still serve old version. It helps out with `getServerSideProps` as well.

- People will also point out that you should generate the most important pages at build time, so that user's won't have to wait as much for these. Sort of, pre-render the 20% of pages that brings about 80% of users. Links (next/link) to the rest of pages, will make those be ISR on the background.

- If you are in charge of the deployment environment, I guess you could keep a previous deploy working, in parallel with the new deploy, and if the new deploy returns 404, you pass that along to the previous deploy. This would be of course, outside the Next.js layer, and sounds like caching with extra steps.

- In favour of having to regenerate all pages, is that, if the API that serves/receives data with these pages, changes, you might see a wide range of behaviours, from the page not updating at all (because ISR errors out), to strange data showing up in the UI, and client side mutations would be unpredictable.

- You could try to minimise the time it takes to generate your page, but that'd involve some trade-offs, keeping as much as possible that's needed for SEO and Time To Visible, and letting the rest be handled by the client.
